hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGrantKycToAccountHandler.java

```diff
@@ -68,6 +69,8 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         if (tokenMeta == null) throw new PreCheckException(INVALID_TOKEN_ID);
         if (tokenMeta.hasKycKey()) {
             context.requireKey(tokenMeta.kycKey());
+        } else {
+            throw new PreCheckException(TOKEN_HAS_NO_KYC_KEY);
```

Comment:
I don't think this is correct. The `validate(...)` method [here](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/token/GrantKycLogic.java#L57) suggests this is not required in precheck